ext/test/http/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ find_package(CURL)
 if(CURL_FOUND)
   set(CURL_LIBRARY "-lcurl")
   set(FILENAME curl_http_test)
+  add_compile_definitions(WITH_CURL)
```

Comment:
Is this needed to apply to all header/cpp source files which include `http_client_factory.h`?

Reply:
Yes, needed for all `cpp` files which include it.  We can add it at root level `CMakeLists.txt`, but probably I felt it would be good to keep that clean. Or if there is better suggestion.

Reply:
Probably we could generate a header file dynamically as `opentelemetry_build_flags`, and put `#define WITH_CURL 1` there when detected, and included the global header file in the above header file or other global place?

Reply:
Thanks for suggestion. If I understood correctly, we need to add the code for dynamic generation of header file within `CMakeFiles`  - Which would be similar to defining a macro ?

Reply:
To define it as macro, I think we need make it global and propagate it to all the consumes which includes the above header file. For dynamic generated header file the option propagation is not an issue because the flag file will always be included.

I think this is not blocking and we could open an issue to track how to make it better.

Reply:
Sure - have created the issue here to trace:: https://github.com/open-telemetry/opentelemetry-cpp/issues/446

Reply:
I reverted it back to original ( and removed default method ) implementation based on comment from Josh : https://github.com/open-telemetry/opentelemetry-cpp/pull/445#discussion_r542423350. 
Let me know if this is not fine.